src/o365/spo/commands/orgassetslibrary/orgassetslibrary-add.ts

```diff
@@ -0,0 +1,140 @@
+    return 'Promotes an existing library to become an organization assets library';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.cdnType = args.options.cdnType || 'Private';
+    telemetryProps.thumbnailUrl = typeof args.options.thumbnailUrl !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    let spoAdminUrl: string = '';
+    const cdnTypeString: string = args.options.cdnType || 'Private';
+    const cdnType: number = cdnTypeString === 'Private' ? 1 : 0;
+    const thumbnailSchema: string = typeof args.options.thumbnailUrl == 'undefined' ? `<Parameter Type="Null" />` : `<Parameter Type="String">${args.options.thumbnailUrl}</Parameter>`;
+
+    this
+      .getSpoAdminUrl(cmd, this.debug)
+      .then((_spoAdminUrl: string): Promise<ContextInfo> => {
+        spoAdminUrl = _spoAdminUrl;
+
+        return this.getRequestDigest(spoAdminUrl);
+      })
+      .then((res: ContextInfo): Promise<string> => {
+        const requestOptions: any = {
+          url: `${spoAdminUrl}/_vti_bin/client.svc/ProcessQuery`,
+          headers: {
+            'X-RequestDigest': res.FormDigestValue
+          },
+          body: `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><Method Name="AddToOrgAssetsLibAndCdnWithType" Id="11" ObjectPathId="8"><Parameters><Parameter Type="Enum">${cdnType}</Parameter><Parameter Type="String">${args.options.libraryUrl}</Parameter>${thumbnailSchema}<Parameter Type="Enum">1</Parameter></Parameters></Method></Actions><ObjectPaths><Constructor Id="8" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}" /></ObjectPaths></Request>`
+        };
+
+        return request.post(requestOptions);
+      })
+      .then((res: string): void => {
+        const json: ClientSvcResponse = JSON.parse(res);
+          const response: ClientSvcResponseContents = json[0];
+          if (response.ErrorInfo) {
+            cb(new CommandError(response.ErrorInfo.ErrorMessage));
+            return;
+          }
+          else {
+            cmd.log(json[json.length - 1]);
+
+            if (this.verbose) {
+              cmd.log(vorpal.chalk.green('DONE'));
+            }
+          }
+          cb();
+      }, (err: any): void => this.handleRejectedPromise(err, cmd, cb));
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      if (!args.options.libraryUrl) {
+        return `Required parameter libraryUrl missing`;
+      }
+
+      if (typeof args.options.thumbnailUrl !== 'undefined' && SpoCommand.isValidSharePointUrl(args.options.thumbnailUrl) != true) {
+        return SpoCommand.isValidSharePointUrl(args.options.thumbnailUrl);
```

Comment:
Should we consider to display friendly string instead of another execution of the validation function?

Reply:
The return value of that function would be `{x} is not a valid SharePoint Online site URL`, do we want to add something else?

Reply:
What bothers me is that we are processing the same thing twice while we can do it once. 

But the performance impact will be negligible, so we can leave it the way it is; or you can run the doublicated function just once before the if statement, store the result in a variable and use it for both the if statement and the return result.